deepchecks/recommender/checks/cold_start_detection.py

```diff
@@ -0,0 +1,129 @@
+            columns: Union[Hashable, List[Hashable], None] = None,
+            ignore_columns: Union[Hashable, List[Hashable], None] = None,
+            n_samples: int = 5_000,
+            n_to_show: int = 10,
+            random_state: int = 42,
+            timeout: int = 10,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.columns = columns
+        self.ignore_columns = ignore_columns
+        self.n_samples = n_samples
+        self.n_to_show = n_to_show
+        
+        self.random_state = random_state
+
+    def run_logic(self, context: Context, dataset_kind) -> CheckResult:
+        """Run check."""
+        dataset = context.get_data_by_kind(dataset_kind)
+        #dataset = dataset.sample(self.n_samples, random_state=self.random_state).drop_na_labels()
+        dataset.assert_datetime()
+
+        df = select_from_dataframe(dataset.data, self.columns, self.ignore_columns)
+        datetime_col = dataset.datetime_name
+        user_col = dataset.user_index_name
+        item_col = dataset.item_index_name
+
+        df = df.sort_values(datetime_col)
+
+        # Group the DataFrame by user
+        grouped_df = df.groupby(user_col)
+
+        # Calculate session length (number of interactions) for each user
+        session_lengths = grouped_df.size()
+
+        # Find the last interaction timestamp for each user
+        last_interaction_timestamps = grouped_df[datetime_col].last()
+        last_interaction_item= grouped_df[item_col].last()
+
+        # Create a new DataFrame with user, session length, and last interaction timestamp
+        extracted_df = pd.DataFrame({
+            f'{user_col}': last_interaction_timestamps.index,
+            'session_length': session_lengths,
+            f'{item_col}': last_interaction_item,
+            f'{datetime_col}': last_interaction_timestamps
+        })
+        
+        cold_start_users = extracted_df[extracted_df['session_length'] == 1]
+        cold_start_users_to_show = cold_start_users.sample(n=self.n_to_show, random_state=42)
+        # Create the check result visualization
+        headnote = """<span>
+                    cold start users refer to new users who have limited or no historical data or interactions within a system or platform.
+                    They have not yet established a significant user history that can be leveraged for personalized recommendations or targeted actions.
+                    For the model, there can be some consequences fromincreased uncertainty to the difficulty in personalization.
+                    From the business side, the consequences are reeduced user engagement, missed opportunities, and limited user understanding.
+                    .<br><br>
+                    </span>"""
+                    
+
+        proportion_cold_start_users = cold_start_users[user_col].nunique()/df[user_col].nunique()
+
+        text = f'{format_percent(proportion_cold_start_users)}% of users are cold start users. '
```

Comment:
A visual display (even a simple one, like a bar or pie chart) will be preferable.

Reply:
fixed.
https://github.com/deepchecks/deepchecks/pull/2610/commits/6c5ee401c6b425ed50cfb89d66303b1260924102